api/src/main/java/io/opentelemetry/correlationcontext/CorrelationContextManager.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.correlationcontext;
+
+import io.opentelemetry.context.Scope;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Object for creating new {@link CorrelationContext}s and {@code CorrelationContext}s based on the
+ * current context.
+ *
+ * <p>This class returns {@link CorrelationContext.Builder builders} that can be used to create the
+ * implementation-dependent {@link CorrelationContext}s.
+ *
+ * <p>Implementations may have different constraints and are free to convert entry contexts to their
+ * own subtypes. This means callers cannot assume the {@link #getCurrentContext() current context}
+ * is the same instance as the one {@link #withContext(CorrelationContext) placed into scope}.
+ *
+ * @since 0.1.0
+ */
+@ThreadSafe
+public interface CorrelationContextManager {
+
+  /**
+   * Returns the current {@code CorrelationContext}.
+   *
+   * @return the current {@code CorrelationContext}.
+   * @since 0.1.0
+   */
+  CorrelationContext getCurrentContext();
```

Comment:
This can be implemented correct? Maybe in a `CorrelationContextUtil`?

Reply:
Sure, lets go with that (after we do other previous steps).

Reply:
On a second read: are you suggesting we remove this method from `CorrelationContextManager`? If that's the case, I'm not sure that's a good idea. Ideally the top layer of each concern (traces, correlations, etc) should provide some sugar ;)